### PR TITLE
feat(api): Download file using UploadID

### PR DESCRIPTION
**Pull request by soham4abc**
## Description

API endpoint for downloading file using Upload ID

### Changes

Created a new function in the `UploadController.php` named `uploadDownload()`

## How to test

use the endpoint `localhost/repo/api/v1/uploads/<UploadID>/download` to get the file as an API response

## Screenshot

 
image

PTAL @GMishx @shaheemazmalmmd 

<a href="https://gitpod.io/#https://github.com/fossology/fossology/pull/2309"></a>

### CODE REVIEW

**Review by GMishx — CHANGES_REQUESTED** [review]

File: src/www/ui/api/Controllers/UploadController.php [hunk 1]
```diff
@@ -237,6 +239,55 @@ public function getUploads($request, $response, $args)
...
+
+    if (isset($args['id'])) {
+      $id = intval($args['id']);
+      $upload = $this->uploadAccessible($this->restHelper->getGroupId(), $id);
+      if ($upload !== true) {
+        return $response->withJson($upload->getArray(), $upload->getCode());
+      }
+    }
+    global $container;
+    $restHelper = $container->get('helper.restHelper');
```
[the remark this answers is not in the document]
Comment (soham4abc) [reply src/www/ui/api/Controllers/UploadController.php:263]: resolved

File: src/www/ui/api/Controllers/UploadController.php [hunk 2]
```diff
@@ -237,6 +239,55 @@ public function getUploads($request, $response, $args)
...
+    $dbManager = $restHelper->getDbHelper()->getDbManager();
```
[the remark this answers is not in the document]
Comment (soham4abc) [reply src/www/ui/api/Controllers/UploadController.php:264]: resolved

File: src/www/ui/api/Controllers/UploadController.php [hunk 3]
```diff
@@ -237,6 +239,55 @@ public function getUploads($request, $response, $args)
...
+    $uploadDao = $restHelper->getUploadDao();
+    $uploadTreeTableName = $uploadDao->getUploadtreeTableName($id);
+    $itemTreeBounds = $uploadDao->getParentItemBounds($id,$uploadTreeTableName);
+    $sql =  "SELECT pfile_fk , ufile_name FROM uploadtree_a WHERE uploadtree_pk=$1";
+    $params = array($itemTreeBounds->getItemId());
+    $params = array($itemTreeBounds->getItemId());
```
[the remark this answers is not in the document]
Comment (soham4abc) [reply src/www/ui/api/Controllers/UploadController.php]: resolved

File: src/www/ui/api/Controllers/UploadController.php [hunk 4]
```diff
@@ -237,6 +239,55 @@ public function getUploads($request, $response, $args)
...
+        $responseContent->getMimeType())
+        ->withHeader('Content-Disposition',
+        $responseFile->headers->get('Content-Disposition'))
+        ->withHeader('Cache-Control', 'must-revalidate')
+        ->withHeader('Pragma', 'private')
+        ->withHeader('Content-Length', filesize($responseContent->getPathname()));
+      $sf = new StreamFactory();
+      $newResponse = $newResponse->withBody(
+        $sf->createStreamFromFile($responseContent->getPathname())
+      );
```
[the remark this answers is not in the document]
Comment (soham4abc) [reply src/www/ui/api/Controllers/UploadController.php]: resolved

**Review by GMishx — APPROVED** [review]
Changes looks good. Tested, working as expected.

**Pull request merged**